### ISSUE TITLE
genomecov does not use read depth from vcf file

### ISSUE THREAD

**Issue by tolot27** [arq5x/bedtools2#602]
If a vcf file is used as input with genomecov (parameter `-i`) only values `0`, `1` and `2` are reported for genomic regions rather than the raw read depth as it is provided in the `INFO` column (`DP=x`) in the vcf file.
Applying genomecov to the underlying bam file (parameter `-i`) reveal the correct read depth values.

**Comment by arq5x** [comment]
Yes, using genomecov on a VCF doesn't really make sense, as a VCF only provides information for polymorphic sites, rather than every site in the genome.

**Comment by tolot27** [comment]
> Yes, using genomecov on a VCF doesn't really make sense, as a VCF only provides information for polymorphic sites, rather than every site in the genome.

I do not agree with your assumption. Just look at the [Samtools Variant Calling Tutorial](https://samtools.github.io/bcftools/howtos/variant-calling.html). There it is written: "`mpileup` part generates genotype likelihoods at each genomic position with coverage." It applies to `samtools mpileup` too.
That is exactly my use case. I don't want to screen the bam file again, if the coverage is already available in the VCF file. As long as genomecov supports the VCF file format as input, it should extract the necessary information if available and fells back, if not.

The user must be aware of that not all VCF files contain all positions.

**Comment by arq5x** [comment]
Fair point, I had forgotten about the mpileup use case. I will look into adding support for the DP field.
